algebird-core/src/main/scala/com/twitter/algebird/Aggregator.scala

```diff
@@ -212,6 +212,16 @@ object Aggregator extends java.io.Serializable {
   def immutableSortedReverseTake[T: Ordering](count: Int): MonoidAggregator[T, TopK[T], Seq[T]] =
     new TopKToListAggregator[T](count)(implicitly[Ordering[T]].reverse)
   /**
+   * Selects exactly 'count' of the input records randomly (or all of the records if there are less
+   * then 'count' total records). This assumes that all 'count' of the records can fit in memory,
+   * so use this only for small values of 'count'.
+   */
+  def randomSample[T](r: java.util.Random)(count: Int): MonoidAggregator[T, PriorityQueue[(Double, T)], Seq[T]] =
```

Comment:
i think this is more commonly called `Reservoir Sampling`: https://en.wikipedia.org/wiki/Reservoir_sampling

random sample is imprecise (could also be constant fraction).

My preference would be for a default (constant) seed version also, and in scala, we generally prefer multiple parameter groups only for functions (or in some esoteric cases where path-dependent types require it). So, can we do:
```def reservoirSample[T](r: java.util.Random, count: Int): ...```